docs/docs/contribute/environment-setup.mdx

```diff
@@ -0,0 +1,82 @@
+2. Fork this repository. On the GitHub repository page, click the "Fork" button to create your own copy of the repository.
+    - Make sure the option 'Copy the `main` branch only' is selected.
+3. Clone the forked repository to your local machine using Git. Execute the following command in your terminal: `git clone [repository URL]`.
+
+> For more insights about forking a repository, refer to the [GitHub Docs - Fork a repo](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) guide.
+
+### Setting up your Local Project
+
+After you've cloned your fork onto your device, navigate to the project directory and execute the following commands to get the project running.
+
+1. Run `npm i` to restore all project dependencies.
+2. Run `npm run build` to build the entire project.
+3. Run `npm link` to create a link/reference to your local project. This allows you to reference your locally installed CLI instead of the npm-hosted package.
+
+That's it! If you run `m365 version` now, you will see that you are using the beta version of CLI for Microsoft 365 in your shell!
+
+:::tip
+
+If you have installed the CLI globally using the `npm i -g @pnp/cli-microsoft365` command, we recommend that you uninstall it before running `npm link`.
+
+:::
+
+## Visual Studio Code extensions
+
+Although you can use any IDE, we recommend using Visual Studio Code. When using VS Code, the following extensions will come in handy:
+
+Extension | Why is it useful?
+--- | ---
+[Mocha Test Explorer](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-mocha-test-adapter) | This extension helps you when writing tests. It can run individual tests and facilitate debugging. You also get a nice overview of all tests within the project.
+[ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) | We use ESLint to ensure code consistency within the project. By installing this extension, you will be notified of problems while writing code.
+
+## Npm Scripts
+
+During your development, you will need a set of commands to test, run, and validate your code. The CLI for Microsoft 365 comes with a set of commands that you can use for each occasion.
+
+Command | Description
+------ | ---
+`npm run watch` | Builds the entire project first. After this, a watcher will trigger an incremental build every time a file is saved. This means that only the changed files are rebuilt, not the entire project. **You typically use this command while developing**.
+`npm run build` | Builds the entire project.
+`npm run clean` | Cleans the output directory by deleting all built files.
+`npm run test` | Runs all tests and checks all ESLint rules. This is a combination of `npm run test:cov` and `npm run lint`. This command is executed in our GitHub workflows when creating a PR.
+`npm run test:cov` | Runs all tests and creates a coverage report.
+<nobr>`npm run test:test`</nobr> | Runs all tests.
+`npm run lint` | Runs all ESLint rules.
```

Comment:
As tests, linting rules and code coverage checks are all executed when you run `npm run test`, I think we can ignore the sub commands that are called. First time contributors really just need to know how to watch, build and run tests IMO.

```suggestion
`npm run test` | Executes all tests and linting rules, also generates a code coverage report. This command is executed automatically when you raise a pull request.
```